utils_nlp/eval/rouge_ext.py

```diff
@@ -0,0 +1,978 @@
+                DUC 2004 have been traited.
+                The solution would be to rewrite the whole perl stemming in python from
+                the original script
+
+        Args:
+          metrics: What ROUGE score to compute. Available: ROUGE-N, ROUGE-L, ROUGE-W.
+              Default: ROUGE-N
+          max_n: N-grams for ROUGE-N if specify. Default:1
+          limit_length: If the summaries must be truncated. Defaut:True
+          length_limit: Number of the truncation where the unit is express int length_limit_Type.
+              Default:665 (bytes)
+          length_limit_type: Unit of length_limit. Available: words, bytes. Default: 'bytes'
+          apply_avg: If we should average the score of multiple samples. Default: True. If
+              apply_Avg & apply_best = False, then each ROUGE scores are independant
+          apply_best: Take the best instead of the average. Default: False, then each ROUGE
+              scores are independant
+          stemming: Apply stemming to summaries. Default: True
+          alpha: Alpha use to compute f1 score: P*R/((1-a)*P + a*R). Default:0.5
+          weight_factor: Weight factor to be used for ROUGE-W. Official rouge score defines
+              it at 1.2. Default: 1.0
+
+        Raises:
+          ValueError: raises exception if metric is not among AVAILABLE_METRICS
+          ValueError: raises exception if length_limit_type is not among
+              AVAILABLE_LENGTH_LIMIT_TYPES
+          ValueError: raises exception if weight_factor < 0
+        """
+        self.metrics = metrics[:] if metrics is not None else RougeExt.DEFAULT_METRICS
+        for m in self.metrics:
+            if m not in RougeExt.AVAILABLE_METRICS:
+                raise ValueError("Unknown metric '{}'".format(m))
+
+        self.max_n = max_n if "rouge-n" in self.metrics else None
+        # Add all rouge-n metrics
+        if self.max_n is not None:
+            index_rouge_n = self.metrics.index("rouge-n")
+            del self.metrics[index_rouge_n]
+            self.metrics += ["rouge-{}".format(n) for n in range(1, self.max_n + 1)]
+        self.metrics = set(self.metrics)
+
+        self.limit_length = limit_length
+        if self.limit_length:
+            if length_limit_type not in RougeExt.AVAILABLE_LENGTH_LIMIT_TYPES:
+                raise ValueError("Unknown length_limit_type '{}'".format(length_limit_type))
+
+        self.length_limit = length_limit
+        if self.length_limit == 0:
+            self.limit_length = False
+        self.length_limit_type = length_limit_type
+        self.stemming = stemming
+
+        self.apply_avg = apply_avg
+        self.apply_best = apply_best
+        self.alpha = alpha
+        self.weight_factor = weight_factor
+        if self.weight_factor <= 0:
+            raise ValueError("ROUGE-W weight factor must greater than 0.")
+
+        self.language = language
+
+        self.sentence_split = RougeExt.SENTENCE_SPLIT_DICT[self.language]
```

Comment:
can we make sentence_split, word_tokenize, remove_char_pattern and stemmer as arguments so that this class can be used for other languages without specifically editing the dictionary inside this class?

Reply:
Agree that makes the code more flexible, but makes the function harder to use for languages we already support. I added the arguments so that the user can provide these if the language is not already supported by the internal dictionary. Please take a look.

Reply:
By "the function harder to use for languages we already support", which function do you mean?  I didn't see the arguments for these functions in the __init__ function.

Reply:
Sorry, I actually meant the RougeExt class, but also the compute_rouge_python function. I added the additional arguments to both. You need to go to the latest commit to see the changes. 
https://github.com/microsoft/nlp-recipes/blob/d4c2a26536a6e0ad5e28a6a34a3736d985b054e3/utils_nlp/eval/rouge/compute_rouge.py#L86
https://github.com/microsoft/nlp-recipes/blob/d4c2a26536a6e0ad5e28a6a34a3736d985b054e3/utils_nlp/eval/rouge/rouge_ext.py#L67